http/src/main/java/io/micronaut/http/multipart/StreamingFileUpload.java

```diff
@@ -79,4 +80,14 @@ default Publisher<Boolean> transferTo(OutputStream outputStream) {
      */
     Publisher<Boolean> delete();
 
+    /**
+     * Create an {@link InputStream} that reads this file. The returned stream must be closed after
+     * use. The stream may block when data isn't yet available.
+     *
+     * @return An {@link InputStream} that reads this file's contents
+     * @since 4.2.0
+     */
+    default InputStream asInputStream() {
```

Comment:
add `@NonNull` to return type

Reply:
annotate with `@Blocking`

Reply:
the method itself is not blocking even though the stream might be. should it still be annotated as such?